packages/remark-stringify/types/index.d.ts

```diff
@@ -1,51 +1,14 @@
 // TypeScript Version: 3.0
 
-import {Compiler, Processor, Plugin} from 'unified'
-import {Node, Parent} from 'unist'
-
-declare class RemarkCompiler implements Compiler {
-  compile(): string
-  visitors: {
-    [key: string]: remarkStringify.Visitor
-  }
-}
+import {Compiler, Plugin} from 'unified'
+import {Options} from 'mdast-util-to-markdown'
 
 declare namespace remarkStringify {
   interface Stringify extends Plugin<[RemarkStringifyOptions?]> {
-    Compiler: typeof RemarkCompiler
-    (this: Processor, options?: RemarkStringifyOptions): void
+    Compiler: Compiler
```

Comment:
double checking, the interface for compiler is unchanged?

Reply:
They used to be classes, now they’re the simpler function version. Similar to rehype. I was under the impression this would be the correct way to type that?

Reply:
I guess I could import `ParserFunction` / `CompilerFunction`?

Reply:
Does it still support classes? Or is it only the function version now?
If it's only function, I'd lean towards pulling in CompilerFunction directly, otherwise Compiler is good.

Reply:
No classes I guess?

These two should be something for unified to interface with though, not really for other users 🤔

Do you have an example of how this would look? rehype-{parse,stringify} also use the function-version, but don‘t use function types: https://github.com/rehypejs/rehype/blob/main/packages/rehype-stringify/types/index.d.ts

Reply:
> If so why is it on the public interface?

No clue! I guess it was already exposed, and so are the rehype ones 🤷‍♂️

Reply:
I pushed to use function versions, but it seems like we could just as well drop them, because there is no such field that can be used?

```
  interface Parse extends Plugin<[RemarkParseOptions?]> {
_    Parser: ParserFunction
  }
```

Reply:
I think we could do that, I think Parser and Compiler used to exist to allow extending the parser.
Since that is no longer the extension point, I think it can be safely be removed :crossed_fingers: